analyzer/src/main/kotlin/managers/utils/MavenSupport.kt

```diff
@@ -671,3 +673,36 @@ class HttpsMirrorSelector(private val originalMirrorSelector: MirrorSelector?) :
         }.build()
     }
 }
+
+/**
+ * A specialized [WorkspaceReader] implementation used when building a Maven project that prevents unnecessary
+ * downloads of binary artifacts.
+ *
+ * When building a Maven project from a pom using Maven's _ProjectBuilder_ API clients have no control over the
+ * downloads of dependencies: If dependencies are to be resolved, all the artifacts of these dependencies are
+ * automatically downloaded. For the purpose of just constructing the dependency tree, this is not needed and only
+ * costs time and bandwidth.
+ *
+ * Unfortunately, there is no official API to prevent the download of dependencies. However, Maven can be tricked to
+ * believe that the artifacts are already present on the local disk - then the download is skipped. This is what
+ * this implementation does. It reports that all binary artifacts are available locally, and only treats poms
+ * correctly, as they may be required for the dependency analysis.
+ */
+private class SkipBinaryDownloadsWorkspaceReader(
+    /** The real workspace reader to delegate to. */
+    val delegate: WorkspaceReader
+) : WorkspaceReader by delegate {
+    /**
+     * Locate the given artifact on the local disk. This implementation does a correct location only for pom files;
+     * for all other artifacts it returns a non-null file. Note: For the purpose of analyzing the project's
+     * dependencies the artifact files are never accessed. Therefore, the concrete file returned here does not
+     * actually matter; it just have to be non-null to indicate that the artifact is present locally.
+     */
+    override fun findArtifact(artifact: Artifact?): File? {
```

Comment:
Upstream docs say `artifact` must not be `null`, so we can use a non-nullable type here (like we do in the other `WorkspaceReader ` implementations).